File: 0944. Delete Columns to Make Sorted/Solution0944.py
from typing import List


class Solution0944:
    def minDeletionSize(self, strs: List[str]) -> int:
        row, col = len(strs), len(strs[0])
        res = 0
        for c in range(col):
            for r in range(1,row):
                if ord(strs[r][c]) < ord(strs[r-1][c]):
                    res += 1
                    break
        return res

sol = Solution0944()
strs = ["cba","daf","ghi"]
res = sol.minDeletionSize(strs)
print('input :', strs)
print('output:', res)

File: 0944. Delete Columns to Make Sorted/Solution0944.cpp
#include<iostream>
#include<vector>
#include<string>

using namespace std;

class Solution0944 {
public:
    int minDeletionSize(vector<string>& strs) {
        int row = strs.size(), col = strs[0].length();
        int res = 0;
        for (int i = 0; i < col; i++) {
            for (int j = 1; j < row; j++) {
                if (strs[j][i] < strs[j-1][i]) {
                    res++;
                    break;
                }
            }
        }
        return res;        
    }
};

int main() {
    vector<string> strs{"cba","daf","ghi"};
    Solution0944 sol = Solution0944();
    int res = sol.minDeletionSize(strs);
    cout << res << endl;
    return -1;
}